mobile/lib/modules/activities/views/activities_page.dart

```diff
@@ -1,307 +1,99 @@
-import 'package:cached_network_image/cached_network_image.dart';
 import 'package:collection/collection.dart';
-import 'package:easy_localization/easy_localization.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_hooks/flutter_hooks.dart' hide Store;
 import 'package:hooks_riverpod/hooks_riverpod.dart';
 import 'package:immich_mobile/extensions/asyncvalue_extensions.dart';
 import 'package:immich_mobile/extensions/build_context_extensions.dart';
 import 'package:immich_mobile/modules/activities/models/activity.model.dart';
 import 'package:immich_mobile/modules/activities/providers/activity.provider.dart';
-import 'package:immich_mobile/shared/models/store.dart';
-import 'package:immich_mobile/shared/ui/confirm_dialog.dart';
-import 'package:immich_mobile/shared/ui/user_circle_avatar.dart';
-import 'package:immich_mobile/extensions/datetime_extensions.dart';
-import 'package:immich_mobile/utils/image_url_builder.dart';
+import 'package:immich_mobile/modules/activities/widgets/activity_text_field.dart';
+import 'package:immich_mobile/modules/activities/widgets/activity_tile.dart';
+import 'package:immich_mobile/modules/activities/widgets/dismissible_activity.dart';
+import 'package:immich_mobile/modules/album/providers/current_album.provider.dart';
+import 'package:immich_mobile/modules/asset_viewer/providers/current_asset.provider.dart';
+import 'package:immich_mobile/shared/providers/user.provider.dart';
 
 class ActivitiesPage extends HookConsumerWidget {
-  final String albumId;
-  final String? assetId;
-  final bool withAssetThumbs;
-  final String appBarTitle;
-  final bool isOwner;
-  final bool isReadOnly;
-  const ActivitiesPage(
-    this.albumId, {
-    this.appBarTitle = "",
-    this.assetId,
-    this.withAssetThumbs = true,
-    this.isOwner = false,
-    this.isReadOnly = false,
+  const ActivitiesPage({
     super.key,
   });
 
   @override
   Widget build(BuildContext context, WidgetRef ref) {
-    final provider =
-        activityStateProvider((albumId: albumId, assetId: assetId));
-    final activities = ref.watch(provider);
-    final inputController = useTextEditingController();
-    final inputFocusNode = useFocusNode();
-    final listViewScrollController = useScrollController();
-    final currentUser = Store.tryGet(StoreKey.currentUser);
-
-    useEffect(
-      () {
-        inputFocusNode.requestFocus();
-        return null;
-      },
-      [],
-    );
-
-    buildTitleWithTimestamp(Activity activity, {bool leftAlign = true}) {
-      final textColor = context.isDarkTheme ? Colors.white : Colors.black;
-      final textStyle = context.textTheme.bodyMedium
-          ?.copyWith(color: textColor.withOpacity(0.6));
+    // Album has to be set in the provider before reaching this page
```

Comment:
which location where the album is set?

Reply:
currentAlbum value is set in the `album_viewer_page`
https://github.com/immich-app/immich/blob/cf76a5cf924d20e9f2f0487d548dc6a0ac575f57/mobile/lib/modules/album/views/album_viewer_page.dart#L41-L45

Similarly, currentAsset is set in the `gallery_viewer`
https://github.com/immich-app/immich/blob/cf76a5cf924d20e9f2f0487d548dc6a0ac575f57/mobile/lib/modules/asset_viewer/views/gallery_viewer.dart#L112-L121

Both are disposed once we leave their respective routes

Reply:
So the non-null operator requires the activity to be opened only from an `album_viewer_page`, or a `gallery_viewer` that comes from within the `album_viewer`. If we try to open an activity view from any other view outside an `album_viewer`, it'll error out.